src/sardana/macroserver/macros/test/sardemoenv.py

```diff
@@ -47,11 +47,19 @@ def init(self, door_name=None):
         if door_name is None:
             door_name = getattr(sardanacustomsettings, 'UNITTEST_DOOR_NAME')
         registerExtensions()
```

Comment:
I think it is better to move the registerExtensions after the door validation, in order to avoid it in case the door is not available.